fs-mock.js

```diff
@@ -126,7 +126,9 @@ MockFs.prototype.open = function (path, flags, charset, options) {
                     charset
                 );
             } else {
-                return new BufferStream(fileNode._chunks, charset);
+                // Clone chunks to avoid side effect
+                var bufferChunks = fileNode._chunks.slice(0);
```

Comment:
Ok fixed slice()

I prefer doing minor change, i can do a separate push for BufferStream.prototype.clone.

Just to validate that what you meant:

```
diff --git i/buffer-stream.js w/buffer-stream.js
index 19184cc..24c356b 100644
--- i/buffer-stream.js
+++ w/buffer-stream.js
@@ -3,7 +3,7 @@ var Q = require("q");
 var Reader = require("./reader");

 module.exports = BufferStream;
-function BufferStream(chunks, charset) {
+function BufferStream(chunks, charset, close) {
     if (!(this instanceof BufferStream)) {
         return new BufferStream(chunks, charset);
     }
@@ -14,7 +14,7 @@ function BufferStream(chunks, charset) {
     }
     this._charset = charset;
     this._chunks = chunks;
-    this._close = Q.defer();
+    this._close = close || Q.defer();
     this.closed = this._close.promise;
 }

@@ -35,6 +35,11 @@ BufferStream.prototype.read = function () {
     return Q.resolve(result);
 };

+BufferStream.prototype.clone = function () {
+    var clone = BufferStream(this._chunks.slice(), this._charset, this._close);
+    return Q.resolve(clone);
+};
+
 BufferStream.prototype.write = function (chunk) {
     if (this._charset) {
         chunk = new Buffer(String(chunk), this._charset);
diff --git i/fs-mock.js w/fs-mock.js
index 84116c6..316fc5c 100644
--- i/fs-mock.js
+++ w/fs-mock.js
@@ -126,9 +126,7 @@ MockFs.prototype.open = function (path, flags, charset, options) {
                     charset
                 );
             } else {
-                // Clone chunks to avoid side effect
-                var bufferChunks = fileNode._chunks.slice();
-                return new BufferStream(bufferChunks, charset);
+                return new BufferStream(bufferChunks, charset).clone();
             }
         }
     });
```